builtins/web/fetch/fetch_event.cpp

```diff
@@ -509,94 +528,58 @@ static void dispatch_fetch_event(HandleObject event, double *total_compute) {
   // LOG("Request handler took %fms\n", diff / 1000);
 }
 
-bool install(api::Engine *engine) {
-  ENGINE = engine;
-  FETCH_HANDLERS = new JS::PersistentRootedObjectVector(engine->cx());
+bool handle_incoming_request(host_api::HttpIncomingRequest * request) {
+  if (!ENGINE->toplevel_evaluated()) {
```

Comment:
What case is this branch for?

Reply:
I extracted this out into a function with a short doc comment explaining the purpose.